addons/survey/models/survey.py

```diff
@@ -0,0 +1,1171 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from collections import Counter, OrderedDict
+import datetime
+from itertools import product
+import logging
+import re
+from urlparse import urljoin
+import uuid
+
+from openerp import api, fields, models, _
+from openerp.addons.website.models.website import slug
+from openerp.exceptions import UserError
+
+
+_logger = logging.getLogger(__name__)
+
+class SurveyStage(models.Model):
+    """Stages for Kanban view of surveys"""
+
+    _name = 'survey.stage'
+    _description = 'Survey Stage'
+    _order = 'sequence,id'
+
+    name = fields.Char(required=True, translate=True)
+    sequence = fields.Integer(default=1)
+    closed = fields.Boolean(default=False, help="If closed, people won't be able to answer to surveys in this column.")
+    fold = fields.Boolean(string="Folded in kanban view")
+
+    _sql_constraints = [
+        ('positive_sequence', 'CHECK(sequence >= 0)', 'Sequence number MUST be a natural')
+    ]
+
+
+class Survey(models.Model):
+    '''Settings for a multi-page/multi-question survey.
+    Each survey can have one or more attached pages, and each page can display
+    one or more questions.
+    '''
+
+    _name = 'survey.survey'
+    _description = 'Survey'
+    _rec_name = 'title'
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    @api.returns('self')
+    def _default_stage(self):
+        return self.env['survey.stage'].search([], limit=1)
+
+    # Model fields #
+
+    title = fields.Char(required=True, translate=True)
+    page_ids = fields.One2many('survey.page', 'survey_id', string='Pages', copy=True)
+    stage_id = fields.Many2one('survey.stage', string="Stage", ondelete="set null", copy=False,
+        default='_default_stage')
+    auth_required = fields.Boolean(string='Login required',
+        help="Users with a public link will be requested to login before taking part to the survey",
+        oldname="authenticate")
+    users_can_go_back = fields.Boolean(help="If checked, users can go back to previous pages.")
```

Comment:
But put ```string``` of such long name is good. B'coz ORM goes and cut this name and generate string.
speed optimize. sorry :wink: